fury/actor.py

```diff
@@ -3954,3 +3956,101 @@ def uncertainty_cone(
     angles = main_dir_uncertainty(evals, evecs, signal, sigma, b_matrix)
 
     return double_cone(centers, evecs, angles, colors, scales, opacity)
+
+
+def texture_on_cube(negx, negy, negz, posx, posy, posz, centers=(0, 0, 0)):
+    """Map RGB or RGBA textures on a cube
+
+    Parameters
+    ----------
+    negx : ndarray
+        Input 2D RGB or RGBA array. Dtype should be uint8.
+    negy : ndarray
+        Input 2D RGB or RGBA array. Dtype should be uint8.
+    negz : ndarray
+        Input 2D RGB or RGBA array. Dtype should be uint8.
+    posx : ndarray
+        Input 2D RGB or RGBA array. Dtype should be uint8.
+    posy : ndarray
+        Input 2D RGB or RGBA array. Dtype should be uint8.
+    posz : ndarray
+        Input 2D RGB or RGBA array. Dtype should be uint8.
+    centers : tuple (3,)
+        The X, Y and Z coordinate of the cube, optional.
+
+         |----|
+         | +Y |
+    |----|----|----|----|
+    | -X | +Z | +X | -Z |
+    |----|----|----|----|
+         | -Y |
+         |----|
+
+    Returns
+    -------
+    actors : list[Actor]
+        A list of Actor objects, one for each face of the cube, in order.
+
+    Implementation
+    --------------
+    Check docs/examples/viz_play_cube.py
+
+    """
+    plane_objects = [PlaneSource() for _ in range(6)]
+
+    center_x, center_y, center_z = centers
+    plane_centers = [
+        (-0.5 + center_x, 0 + center_y, 0 + center_z),
+        (0 + center_x, -0.5 + center_y, 0 + center_z),
+        (0 + center_x, 0 + center_y, -0.5 + center_z),
+        (0.5 + center_x, 0 + center_y, 0 + center_z),
+        (0 + center_x, 0.5 + center_y, 0 + center_z),
+        (0 + center_x, 0 + center_y, 0.5 + center_z)
+    ]
+
+    plane_normals = [
+        (-1, 0, 0),
+        (0, -1, 0),
+        (0, 0, -1),
+        (1, 0, 0),
+        (0, 1, 0),
+        (0, 0, 1)
+    ]
+
+    for plane, center, normal in zip(
+        plane_objects,
+        plane_centers,
+        plane_normals
+    ):
+        plane.SetCenter(*center)
+        plane.SetNormal(*normal)
+
+    image_grids = [negx, negy, negz, posx, posy, posz]
+    image_data_objects = [
+        numpy_to_vtk_image_data(grid) for grid in image_grids
+    ]
```

Comment:
is it crashing if not uint8 ?  if yes, you need to check and raise a `ValueError`if there is any issue

Reply:
`uint8` constraint removed. 

Resolved.